hls-test-utils/src/Test/Hls.hs

```diff
@@ -447,6 +451,26 @@ runSessionWithServerInTmpDir' plugins conf sessConf caps tree act = withLock loc
     _fs <- FS.materialiseVFT tmpDir tree
     runSessionWithServer' plugins conf sessConf caps tmpDir act
 
+-- | Setup the test environment for isolated tests.
+--
+-- This creates a directory in the temporary directory that will be
+-- reused for running isolated tests.
+-- It returns the root to the testing directory that tests should use.
+-- This directory is not fully cleaned between reruns.
+-- However, it is totally safe to delete the directory between runs.
+--
+-- Additionally, this overwrites the 'XDG_CACHE_HOME' variable to isolate
+-- the tests from existing caches. 'hie-bios' and 'ghcide' honour the
+-- 'XDG_CACHE_HOME' environment variable and generate their caches there.
+setupTestEnvironment :: IO FilePath
+setupTestEnvironment = do
+  tmpDirRoot <- getTemporaryDirectory
+  let testRoot = tmpDirRoot </> "hls-test-root"
+      testCacheDir = testRoot </> ".cache"
+  createDirectoryIfMissing False testRoot
+  createDirectoryIfMissing False testCacheDir
```

Comment:
Nitpick: I think you can replace these 2 with just 
```suggestion
  createDirectoryIfMissing True testCacheDir
```

Reply:
I wanted to make sure that `$TMPDIR` exists on disk before running the tests (`getTemporaryDirectory` honour the `$TMPDIR` environment variable). I am not sure why I think that's a good idea, if you think this is unhelpful, I am happy to update :)

Reply:
I don't insist, keep it as is if you think it's better in some way.